pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -384,6 +405,136 @@ func (r *ReconcileComplianceScan) phaseLaunchingHandler(h scanTypeHandler, logge
 	return reconcile.Result{}, nil
 }
 
+// getRuntimeKubeletConfig gets the runtime kubeletconfig for a node
+// by fetching /api/v1/nodes/$nodeName/proxy/configz endpoint use the kubeClientset
+// and store it in a configmap for each node
+func (r *ReconcileComplianceScan) getRuntimeKubeletConfig(nodeName string) (string, error) {
+	// get the runtime kubeletconfig
+	kubeletConfigIO, err := r.ClientSet.CoreV1().RESTClient().Get().RequestURI("/api/v1/nodes/" + nodeName + "/proxy/configz").Stream(context.TODO())
+	if err != nil {
+		return "", err
+	}
+	defer kubeletConfigIO.Close()
+	kubeletConfig, err := ioutil.ReadAll(kubeletConfigIO)
+	if err != nil {
+		return "", err
+	}
+	// kubeletConfig is a byte array, we need to convert it to string
+	kubeletConfigStr := string(kubeletConfig)
+	return kubeletConfigStr, nil
+}
+
+func (r *ReconcileComplianceScan) createKubeletConfigCM(instance *compv1alpha1.ComplianceScan, node *corev1.Node, kubeletConfigStr string) error {
```

Comment:
It doesn't look like the last argument is used anywhere (`kubeletConfigStr`). Can we remove it?

Reply:
Good catch!